src/devices/vdml_device.cpp

```diff
@@ -28,18 +28,40 @@ bool Device::is_installed() {
 	return_port(zero_indexed_port, _deviceType == plugged_device_type);
 }
 
-std::uint8_t Device::get_port(void) {
+std::uint8_t Device::get_port(void) const {
 	return _port;
 }
 
 pros::DeviceType Device::get_plugged_type() const {
-	if (!port_mutex_take(_port - 1)) {                            
+	return(get_plugged_type(_port));
+}
+
+pros::DeviceType Device::get_plugged_type(std::uint8_t port) {
+	if (!port_mutex_take(port - 1)) {                            
 		errno = EACCES; 
 		return DeviceType::undefined;                                                                           
 	}
-	DeviceType type = (DeviceType) pros::c::registry_get_plugged_type(_port - 1);
+	DeviceType type = (DeviceType) pros::c::registry_get_plugged_type(port - 1);
 	
-	return_port(_port - 1, type);
+	return_port(port - 1, type);
+}
+
+std::vector<Device> Device::get_all_devices(pros::DeviceType device_type) {
+	std::vector<Device> device_list {};
+
+	for (std::uint8_t curr_port = 0; curr_port < 21; ++curr_port) {
```

Comment:
Why not have curr_port be std::int8_t and then remove the cast on line 60?

Reply:
The cast is to a uint8_t on line 60 because of type promotion to an int when I did a uint8_t value + 1. The Device constructor expects a uint8_t.

Reply:
Ok that makes sense then